torch/_dynamo/utils.py

```diff
@@ -439,6 +439,9 @@ def hashable(x):
     # cannot hash writable memoryview object
     except ValueError:
         return False
+    # not fully initialized object
+    except AttributeError:
+        return False
```

Comment:
`obj.realize()` calls `hash(obj)` which will fail here becasue the `__init__` method is not returned yet.

Reply:
Why are we calling realize on an uninitalized object?

Reply:
`isinstance(obj, UserDefinedObjectVariable)` calls `__isinstance_check__`. That realizes the lazy object and calls `hash(obj)`

Reply:
Where is `isinstance(obj, UserDefinedObjectVariable)` being called on an uninitalized object?  Do you have a full stacktrace?

Reply:
Full log:

<details>
<summary>`TORCH_LOGS="+dynamo" TORCHDYNAMO_VERBOSE=1 python test/dynamo/test_misc.py -k test_object_setattr`</summary>

```console
$ TORCH_LOGS="+dynamo" TORCHDYNAMO_VERBOSE=1 python test/dynamo/test_misc.py -k test_object_setattr                   
/home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py:9732: SyntaxWarning: invalid escape sequence '\.'
  "^This compiled backward function is being run with torch\.use_deterministic_algorithms",
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0] torchdynamo start compiling fn1 /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py:3066, stack (elided 5 frames):
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py", line 10357, in <module>
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     run_tests()
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/test_case.py", line 35, in run_tests
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     run_tests()
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Projects/pytorch/torch/testing/_internal/common_utils.py", line 1167, in run_tests
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     unittest.main(argv=argv)
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/main.py", line 105, in __init__
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     self.runTests()
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/main.py", line 281, in runTests
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     self.result = testRunner.run(self.test)
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/runner.py", line 240, in run
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     test(result)
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/suite.py", line 84, in __call__
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     return self.run(*args, **kwds)
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/suite.py", line 122, in run
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     test(result)
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/suite.py", line 84, in __call__
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     return self.run(*args, **kwds)
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/suite.py", line 122, in run
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     test(result)
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/case.py", line 690, in __call__
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     return self.run(*args, **kwds)
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Projects/pytorch/torch/testing/_internal/common_utils.py", line 2868, in run
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     self._run_custom(
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Projects/pytorch/torch/testing/_internal/common_utils.py", line 2840, in _run_custom
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     super_run(result=result)
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/case.py", line 634, in run
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     self._callTestMethod(testMethod)
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/case.py", line 589, in _callTestMethod
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     if method() is not None:
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Projects/pytorch/torch/testing/_internal/common_utils.py", line 2741, in wrapper
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     method(*args, **kwargs)
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py", line 3076, in test_object_setattr
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     obj12 = opt_fn1(x1.clone())
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/eval_frame.py", line 410, in _fn
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     return fn(*args, **kwargs)
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]   File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 978, in catch_errors
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0]     return callback(frame, cache_entry, hooks, frame_state, skip=1)
V0416 15:25:38.528000 139970242536640 torch/_dynamo/convert_frame.py:676] [0/0] 
I0416 15:25:38.529000 139970242536640 torch/_dynamo/logging.py:55] [0/0] Step 1: torchdynamo start tracing fn1 /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py:3066
V0416 15:25:38.529000 139970242536640 torch/fx/experimental/symbolic_shapes.py:2049] [0/0] create_env
V0416 15:25:38.536000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE COPY_FREE_VARS 1 []
V0416 15:25:38.536000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [0/0] [__trace_source] TRACE starts_line /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py:3066 in fn1 (MiscTests.test_object_setattr.fn1)
V0416 15:25:38.536000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [0/0] [__trace_source]             def fn1(x) -> None:
V0416 15:25:38.681000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE RESUME 0 []
V0416 15:25:38.682000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [0/0] [__trace_source] TRACE starts_line /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py:3067 in fn1 (MiscTests.test_object_setattr.fn1)
V0416 15:25:38.682000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [0/0] [__trace_source]                 a = A(x)
V0416 15:25:38.682000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE PUSH_NULL None []
V0416 15:25:38.682000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE LOAD_DEREF A [NullVariable()]
V0416 15:25:38.682000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE LOAD_FAST x [NullVariable(), LazyVariableTracker()]
V0416 15:25:38.682000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE CALL 1 [NullVariable(), LazyVariableTracker(), LazyVariableTracker()]
V0416 15:25:38.683000 139970242536640 torch/_dynamo/output_graph.py:2046] [0/0] create_graph_input L_x_ L['x']
V0416 15:25:38.683000 139970242536640 torch/_dynamo/variables/builder.py:1975] [0/0] wrap_to_fake L['x'] (10,) StatefulSymbolicContext(dynamic_sizes=[<DimDynamic.STATIC: 2>], constraint_sizes=[None], view_base_context=None, tensor_source=LocalSource(local_name='x', cell_or_freevar=False), shape_env_to_source_to_symbol_cache={}) <class 'torch.Tensor'>
V0416 15:25:38.684000 139970242536640 torch/_dynamo/symbolic_convert.py:2470] [0/0] [__trace_call] TRACE inlined call __init__ from /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py:3067 in fn1 (MiscTests.test_object_setattr.fn1)
V0416 15:25:38.684000 139970242536640 torch/_dynamo/symbolic_convert.py:2470] [0/0] [__trace_call]             a = A(x)
V0416 15:25:38.684000 139970242536640 torch/_dynamo/symbolic_convert.py:2470] [0/0] [__trace_call]                 ~^^^
V0416 15:25:38.684000 139970242536640 torch/_dynamo/symbolic_convert.py:2471] [0/0] INLINING <code object __init__ at 0x7f4d226165d0, file "<string>", line 2>, inlined according trace_rules.lookup inlined by default
V0416 15:25:38.684000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [0/0] [__trace_source] TRACE starts_line <string>:2 in __init__ (inline depth: 1)
V0416 15:25:38.684000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [0/0] [__trace_source]     
V0416 15:25:38.684000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE RESUME 0 []
V0416 15:25:38.685000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [0/0] [__trace_source] TRACE starts_line <string>:3 in __init__ (inline depth: 1)
V0416 15:25:38.685000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [0/0] [__trace_source]     
V0416 15:25:38.685000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE LOAD_FAST x []
V0416 15:25:38.685000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE LOAD_FAST self [TensorVariable()]
V0416 15:25:38.685000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE STORE_ATTR x [TensorVariable(), UserDefinedObjectVariable()]
V0416 15:25:38.685000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE RETURN_CONST None []
V0416 15:25:38.685000 139970242536640 torch/_dynamo/symbolic_convert.py:2516] [0/0] DONE INLINING <code object __init__ at 0x7f4d226165d0, file "<string>", line 2>
V0416 15:25:38.685000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE STORE_FAST a [UserDefinedObjectVariable()]
V0416 15:25:38.685000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [0/0] [__trace_source] TRACE starts_line /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py:3068 in fn1 (MiscTests.test_object_setattr.fn1)
V0416 15:25:38.685000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [0/0] [__trace_source]                 object.__setattr__(a, "x", x + 2)
V0416 15:25:38.685000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE LOAD_GLOBAL object []
V0416 15:25:38.685000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE LOAD_ATTR __setattr__ [BuiltinVariable()]
V0416 15:25:38.685000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE LOAD_FAST a [NullVariable(), GetAttrVariable()]
V0416 15:25:38.685000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE LOAD_CONST x [NullVariable(), GetAttrVariable(), UserDefinedObjectVariable()]
V0416 15:25:38.685000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE LOAD_FAST x [NullVariable(), GetAttrVariable(), UserDefinedObjectVariable(), ConstantVariable()]
V0416 15:25:38.686000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE LOAD_CONST 2 [NullVariable(), GetAttrVariable(), UserDefinedObjectVariable(), ConstantVariable(), TensorVariable()]
V0416 15:25:38.686000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE BINARY_OP 0 [NullVariable(), GetAttrVariable(), UserDefinedObjectVariable(), ConstantVariable(), TensorVariable(), ConstantVariable()]
V0416 15:25:38.686000 139970242536640 torch/_dynamo/output_graph.py:1905] [0/0] [__trace_call] TRACE FX call add from /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py:3068 in fn1 (MiscTests.test_object_setattr.fn1)
V0416 15:25:38.686000 139970242536640 torch/_dynamo/output_graph.py:1905] [0/0] [__trace_call]             object.__setattr__(a, "x", x + 2)
V0416 15:25:38.686000 139970242536640 torch/_dynamo/output_graph.py:1905] [0/0] [__trace_call]                                        ~~^~~
V0416 15:25:38.687000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE CALL 3 [NullVariable(), GetAttrVariable(), UserDefinedObjectVariable(), ConstantVariable(), TensorVariable()]
V0416 15:25:38.687000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE POP_TOP None [ConstantVariable()]
V0416 15:25:38.687000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [0/0] [__trace_source] TRACE starts_line /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py:3069 in fn1 (MiscTests.test_object_setattr.fn1)
V0416 15:25:38.687000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [0/0] [__trace_source]                 return a
V0416 15:25:38.687000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE LOAD_FAST a []
V0416 15:25:38.687000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [0/0] TRACE RETURN_VALUE None [UserDefinedObjectVariable()]
I0416 15:25:38.687000 139970242536640 torch/_dynamo/logging.py:55] [0/0] Step 1: torchdynamo done tracing fn1 (RETURN_VALUE)
V0416 15:25:38.687000 139970242536640 torch/_dynamo/symbolic_convert.py:2343] [0/0] RETURN_VALUE triggered compile
V0416 15:25:38.688000 139970242536640 torch/_dynamo/output_graph.py:933] [0/0] COMPILING GRAPH due to GraphCompileReason(reason='return_value', user_stack=[<FrameSummary file /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py, line 3069 in fn1>], graph_break=False)
V0416 15:25:38.688000 139970242536640 torch/_dynamo/output_graph.py:1240] [0/0] [__graph_code] TRACED GRAPH
V0416 15:25:38.688000 139970242536640 torch/_dynamo/output_graph.py:1240] [0/0] [__graph_code]  ===== __compiled_fn_1 =====
V0416 15:25:38.688000 139970242536640 torch/_dynamo/output_graph.py:1240] [0/0] [__graph_code]  /home/PanXuehai/Projects/pytorch/torch/fx/_lazy_graph_module.py class GraphModule(torch.nn.Module):
V0416 15:25:38.688000 139970242536640 torch/_dynamo/output_graph.py:1240] [0/0] [__graph_code]     def forward(self, L_x_ : torch.Tensor):
V0416 15:25:38.688000 139970242536640 torch/_dynamo/output_graph.py:1240] [0/0] [__graph_code]         l_x_ = L_x_
V0416 15:25:38.688000 139970242536640 torch/_dynamo/output_graph.py:1240] [0/0] [__graph_code]         
V0416 15:25:38.688000 139970242536640 torch/_dynamo/output_graph.py:1240] [0/0] [__graph_code]         # File: /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py:3068 in fn1, code: object.__setattr__(a, "x", x + 2)
V0416 15:25:38.688000 139970242536640 torch/_dynamo/output_graph.py:1240] [0/0] [__graph_code]         add = l_x_ + 2;  l_x_ = None
V0416 15:25:38.688000 139970242536640 torch/_dynamo/output_graph.py:1240] [0/0] [__graph_code]         return (add,)
V0416 15:25:38.688000 139970242536640 torch/_dynamo/output_graph.py:1240] [0/0] [__graph_code]         
V0416 15:25:38.688000 139970242536640 torch/_dynamo/output_graph.py:1240] [0/0] [__graph_code] 
V0416 15:25:38.689000 139970242536640 torch/_dynamo/output_graph.py:1246] [0/0] [__graph] Tabulate module missing, please install tabulate to log the graph in tabular format, logging code instead:
V0416 15:25:38.689000 139970242536640 torch/_dynamo/output_graph.py:1246] [0/0] [__graph] TRACED GRAPH
V0416 15:25:38.689000 139970242536640 torch/_dynamo/output_graph.py:1246] [0/0] [__graph]  ===== __compiled_fn_1 =====
V0416 15:25:38.689000 139970242536640 torch/_dynamo/output_graph.py:1246] [0/0] [__graph]  /home/PanXuehai/Projects/pytorch/torch/fx/_lazy_graph_module.py class GraphModule(torch.nn.Module):
V0416 15:25:38.689000 139970242536640 torch/_dynamo/output_graph.py:1246] [0/0] [__graph]     def forward(self, L_x_ : torch.Tensor):
V0416 15:25:38.689000 139970242536640 torch/_dynamo/output_graph.py:1246] [0/0] [__graph]         l_x_ = L_x_
V0416 15:25:38.689000 139970242536640 torch/_dynamo/output_graph.py:1246] [0/0] [__graph]         
V0416 15:25:38.689000 139970242536640 torch/_dynamo/output_graph.py:1246] [0/0] [__graph]         # File: /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py:3068 in fn1, code: object.__setattr__(a, "x", x + 2)
V0416 15:25:38.689000 139970242536640 torch/_dynamo/output_graph.py:1246] [0/0] [__graph]         add = l_x_ + 2;  l_x_ = None
V0416 15:25:38.689000 139970242536640 torch/_dynamo/output_graph.py:1246] [0/0] [__graph]         return (add,)
V0416 15:25:38.689000 139970242536640 torch/_dynamo/output_graph.py:1246] [0/0] [__graph]         
V0416 15:25:38.689000 139970242536640 torch/_dynamo/output_graph.py:1246] [0/0] [__graph] 
V0416 15:25:38.689000 139970242536640 torch/_dynamo/output_graph.py:1247] [0/0] [__graph_sizes] TRACED GRAPH TENSOR SIZES
V0416 15:25:38.689000 139970242536640 torch/_dynamo/output_graph.py:1247] [0/0] [__graph_sizes] ===== __compiled_fn_1 =====
V0416 15:25:38.689000 139970242536640 torch/_dynamo/output_graph.py:1247] [0/0] [__graph_sizes] l_x_: (10,)
V0416 15:25:38.689000 139970242536640 torch/_dynamo/output_graph.py:1247] [0/0] [__graph_sizes] add: (10,)
V0416 15:25:38.689000 139970242536640 torch/_dynamo/output_graph.py:1247] [0/0] [__graph_sizes] 
I0416 15:25:38.689000 139970242536640 torch/_dynamo/logging.py:55] [0/0] Step 2: calling compiler function debug_wrapper
I0416 15:25:38.690000 139970242536640 torch/_dynamo/logging.py:55] [0/0] Step 2: done compiler function debug_wrapper
I0416 15:25:38.692000 139970242536640 torch/fx/experimental/symbolic_shapes.py:2921] [0/0] produce_guards
V0416 15:25:38.692000 139970242536640 torch/fx/experimental/symbolic_shapes.py:3103] [0/0] track_symint L['x'].size()[0] 10 None
V0416 15:25:38.692000 139970242536640 torch/fx/experimental/symbolic_shapes.py:3103] [0/0] track_symint L['x'].stride()[0] 1 None
V0416 15:25:38.692000 139970242536640 torch/fx/experimental/symbolic_shapes.py:3103] [0/0] track_symint L['x'].storage_offset() 0 None
V0416 15:25:38.692000 139970242536640 torch/fx/experimental/symbolic_shapes.py:3249] [0/0] Skipping guard L['x'].size()[0] == 10
V0416 15:25:38.692000 139970242536640 torch/fx/experimental/symbolic_shapes.py:3249] [0/0] Skipping guard L['x'].stride()[0] == 1
V0416 15:25:38.692000 139970242536640 torch/fx/experimental/symbolic_shapes.py:3249] [0/0] Skipping guard L['x'].storage_offset() == 0
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1789] [0/0] [__guards] GUARDS:
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] 
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] TREE_GUARD_MANAGER:
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] +- RootGuardManager
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] | +- DEFAULT_DEVICE: utils_device.CURRENT_DEVICE == None                           # _dynamo/output_graph.py:452 in init_ambient_guards
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] | +- GLOBAL_STATE: ___check_global_state()
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] | +- GuardManager: source=L['A'], accessed_by=DictGetItemGuardAccessor(A)
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] | | +- ID_MATCH: ___check_obj_id(L['A'], 129514816)                            # a = A(x)  # test/dynamo/test_misc.py:3067 in fn1
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] | +- GuardManager: source=L['x'], accessed_by=DictGetItemGuardAccessor(x)
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] | | +- TENSOR_MATCH: check_tensor(L['x'], Tensor, DispatchKeySet(CPU, BackendSelect, ADInplaceOrView, AutogradCPU), torch.float32, device=None, requires_grad=False, size=[10], stride=[1])  # a = A(x)  # test/dynamo/test_misc.py:3067 in fn1
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] | | +- NO_HASATTR: hasattr(L['x'], '_dynamo_dynamic_indices') == False           # a = A(x)  # test/dynamo/test_misc.py:3067 in fn1
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] | +- GuardManager: source=G, accessed_by=GlobalsGuardAccessor
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] | | +- DictGuardManager: source=G['__builtins_dict___0'], accessed_by=DictGetItemGuardAccessor(__builtins_dict___0)
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] | | | +- KeyValueManager pair at index=69
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] | | | | +- KeyManager: GuardManager: source=list(G['__builtins_dict___0'].keys())[69]
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] | | | | | +- EQUALS_MATCH: list(G['__builtins_dict___0'].keys())[69] == 'object'
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] | | | | +- ValueManager: GuardManager: source=G['__builtins_dict___0']['object']
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] | | | | | +- ID_MATCH: ___check_obj_id(G['__builtins_dict___0']['object'], 9721312)  # object.__setattr__(a, "x", x + 2)  # test/dynamo/test_misc.py:3068 in fn1
V0416 15:25:38.693000 139970242536640 torch/_dynamo/guards.py:1766] [0/0] [__guards] 
V0416 15:25:38.694000 139970242536640 torch/_dynamo/convert_frame.py:948] skipping: _fn (reason: in skipfiles, file: /home/PanXuehai/Projects/pytorch/torch/_dynamo/eval_frame.py)
V0416 15:25:38.694000 139970242536640 torch/_dynamo/convert_frame.py:948] skipping: always_false (reason: in skipfiles, file: /home/PanXuehai/Projects/pytorch/torch/_dynamo/eval_frame.py)
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0] torchdynamo start compiling fn2 /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py:3086, stack (elided 5 frames):
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py", line 10357, in <module>
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     run_tests()
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/test_case.py", line 35, in run_tests
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     run_tests()
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Projects/pytorch/torch/testing/_internal/common_utils.py", line 1167, in run_tests
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     unittest.main(argv=argv)
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/main.py", line 105, in __init__
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     self.runTests()
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/main.py", line 281, in runTests
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     self.result = testRunner.run(self.test)
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/runner.py", line 240, in run
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     test(result)
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/suite.py", line 84, in __call__
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     return self.run(*args, **kwds)
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/suite.py", line 122, in run
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     test(result)
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/suite.py", line 84, in __call__
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     return self.run(*args, **kwds)
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/suite.py", line 122, in run
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     test(result)
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/case.py", line 690, in __call__
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     return self.run(*args, **kwds)
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Projects/pytorch/torch/testing/_internal/common_utils.py", line 2868, in run
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     self._run_custom(
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Projects/pytorch/torch/testing/_internal/common_utils.py", line 2840, in _run_custom
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     super_run(result=result)
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/case.py", line 634, in run
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     self._callTestMethod(testMethod)
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/case.py", line 589, in _callTestMethod
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     if method() is not None:
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Projects/pytorch/torch/testing/_internal/common_utils.py", line 2741, in wrapper
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     method(*args, **kwargs)
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py", line 3095, in test_object_setattr
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     obj22 = opt_fn2(x2.clone())
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/eval_frame.py", line 410, in _fn
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     return fn(*args, **kwargs)
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]   File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 978, in catch_errors
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0]     return callback(frame, cache_entry, hooks, frame_state, skip=1)
V0416 15:25:38.695000 139970242536640 torch/_dynamo/convert_frame.py:676] [1/0] 
I0416 15:25:38.696000 139970242536640 torch/_dynamo/logging.py:55] [1/0] Step 1: torchdynamo start tracing fn2 /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py:3086
V0416 15:25:38.696000 139970242536640 torch/fx/experimental/symbolic_shapes.py:2049] [1/0] create_env
V0416 15:25:38.696000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE COPY_FREE_VARS 1 []
V0416 15:25:38.696000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [1/0] [__trace_source] TRACE starts_line /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py:3086 in fn2 (MiscTests.test_object_setattr.fn2)
V0416 15:25:38.696000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [1/0] [__trace_source]             def fn2(x) -> None:
V0416 15:25:38.696000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE RESUME 0 []
V0416 15:25:38.696000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [1/0] [__trace_source] TRACE starts_line /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py:3087 in fn2 (MiscTests.test_object_setattr.fn2)
V0416 15:25:38.696000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [1/0] [__trace_source]                 b = B(x)
V0416 15:25:38.696000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE PUSH_NULL None []
V0416 15:25:38.697000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE LOAD_DEREF B [NullVariable()]
V0416 15:25:38.697000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE LOAD_FAST x [NullVariable(), LazyVariableTracker()]
V0416 15:25:38.697000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE CALL 1 [NullVariable(), LazyVariableTracker(), LazyVariableTracker()]
V0416 15:25:38.697000 139970242536640 torch/_dynamo/output_graph.py:2046] [1/0] create_graph_input L_x_ L['x']
V0416 15:25:38.697000 139970242536640 torch/_dynamo/variables/builder.py:1975] [1/0] wrap_to_fake L['x'] (10,) StatefulSymbolicContext(dynamic_sizes=[<DimDynamic.STATIC: 2>], constraint_sizes=[None], view_base_context=None, tensor_source=LocalSource(local_name='x', cell_or_freevar=False), shape_env_to_source_to_symbol_cache={}) <class 'torch.Tensor'>
V0416 15:25:38.698000 139970242536640 torch/_dynamo/symbolic_convert.py:2470] [1/0] [__trace_call] TRACE inlined call __init__ from /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py:3087 in fn2 (MiscTests.test_object_setattr.fn2)
V0416 15:25:38.698000 139970242536640 torch/_dynamo/symbolic_convert.py:2470] [1/0] [__trace_call]             b = B(x)
V0416 15:25:38.698000 139970242536640 torch/_dynamo/symbolic_convert.py:2470] [1/0] [__trace_call]                 ~^^^
V0416 15:25:38.698000 139970242536640 torch/_dynamo/symbolic_convert.py:2471] [1/0] INLINING <code object __init__ at 0x7f4d225135a0, file "<string>", line 2>, inlined according trace_rules.lookup inlined by default
V0416 15:25:38.698000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE COPY_FREE_VARS 1 []
V0416 15:25:38.698000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [1/0] [__trace_source] TRACE starts_line <string>:2 in __init__ (inline depth: 1)
V0416 15:25:38.698000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [1/0] [__trace_source]     
V0416 15:25:38.698000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE RESUME 0 []
V0416 15:25:38.698000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [1/0] [__trace_source] TRACE starts_line <string>:3 in __init__ (inline depth: 1)
V0416 15:25:38.698000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [1/0] [__trace_source]     
V0416 15:25:38.698000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE LOAD_DEREF __dataclass_builtins_object__ []
V0416 15:25:38.698000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE LOAD_ATTR __setattr__ [BuiltinVariable()]
V0416 15:25:38.699000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE LOAD_FAST self [NullVariable(), GetAttrVariable()]
V0416 15:25:38.699000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE LOAD_CONST x [NullVariable(), GetAttrVariable(), UserDefinedObjectVariable()]
V0416 15:25:38.699000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE LOAD_FAST x [NullVariable(), GetAttrVariable(), UserDefinedObjectVariable(), ConstantVariable()]
V0416 15:25:38.699000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE CALL 3 [NullVariable(), GetAttrVariable(), UserDefinedObjectVariable(), ConstantVariable(), TensorVariable()]
V0416 15:25:38.699000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE POP_TOP None [ConstantVariable()]
V0416 15:25:38.699000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE RETURN_CONST None []
V0416 15:25:38.699000 139970242536640 torch/_dynamo/symbolic_convert.py:2516] [1/0] DONE INLINING <code object __init__ at 0x7f4d225135a0, file "<string>", line 2>
V0416 15:25:38.699000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE STORE_FAST b [UserDefinedObjectVariable()]
V0416 15:25:38.699000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [1/0] [__trace_source] TRACE starts_line /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py:3088 in fn2 (MiscTests.test_object_setattr.fn2)
V0416 15:25:38.699000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [1/0] [__trace_source]                 return b
V0416 15:25:38.699000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE LOAD_FAST b []
V0416 15:25:38.699000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [1/0] TRACE RETURN_VALUE None [UserDefinedObjectVariable()]
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:582] [1/0] Skipping frame because no content in function call fn2                     /home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py 3086
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0] torchdynamo start compiling __init__ <string>:2, stack (elided 5 frames):
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py", line 10357, in <module>
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     run_tests()
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/test_case.py", line 35, in run_tests
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     run_tests()
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Projects/pytorch/torch/testing/_internal/common_utils.py", line 1167, in run_tests
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     unittest.main(argv=argv)
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/main.py", line 105, in __init__
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     self.runTests()
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/main.py", line 281, in runTests
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     self.result = testRunner.run(self.test)
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/runner.py", line 240, in run
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     test(result)
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/suite.py", line 84, in __call__
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     return self.run(*args, **kwds)
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/suite.py", line 122, in run
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     test(result)
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/suite.py", line 84, in __call__
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     return self.run(*args, **kwds)
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/suite.py", line 122, in run
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     test(result)
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/case.py", line 690, in __call__
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     return self.run(*args, **kwds)
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Projects/pytorch/torch/testing/_internal/common_utils.py", line 2868, in run
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     self._run_custom(
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Projects/pytorch/torch/testing/_internal/common_utils.py", line 2840, in _run_custom
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     super_run(result=result)
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/case.py", line 634, in run
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     self._callTestMethod(testMethod)
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/unittest/case.py", line 589, in _callTestMethod
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     if method() is not None:
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Projects/pytorch/torch/testing/_internal/common_utils.py", line 2741, in wrapper
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     method(*args, **kwargs)
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py", line 3095, in test_object_setattr
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     obj22 = opt_fn2(x2.clone())
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/eval_frame.py", line 410, in _fn
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     return fn(*args, **kwargs)
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py", line 3087, in fn2
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     b = B(x)
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]   File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 978, in catch_errors
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0]     return callback(frame, cache_entry, hooks, frame_state, skip=1)
V0416 15:25:38.699000 139970242536640 torch/_dynamo/convert_frame.py:676] [2/0] 
I0416 15:25:38.700000 139970242536640 torch/_dynamo/logging.py:55] [2/0] Step 1: torchdynamo start tracing __init__ <string>:2
V0416 15:25:38.700000 139970242536640 torch/fx/experimental/symbolic_shapes.py:2049] [2/0] create_env
V0416 15:25:38.700000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [2/0] TRACE COPY_FREE_VARS 1 []
V0416 15:25:38.700000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [2/0] [__trace_source] TRACE starts_line <string>:2 in __init__
V0416 15:25:38.700000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [2/0] [__trace_source]     
V0416 15:25:38.700000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [2/0] TRACE RESUME 0 []
V0416 15:25:38.701000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [2/0] [__trace_source] TRACE starts_line <string>:3 in __init__
V0416 15:25:38.701000 139970242536640 torch/_dynamo/symbolic_convert.py:761] [2/0] [__trace_source]     
V0416 15:25:38.701000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [2/0] TRACE LOAD_DEREF __dataclass_builtins_object__ []
V0416 15:25:38.701000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [2/0] TRACE LOAD_ATTR __setattr__ [LazyVariableTracker()]
V0416 15:25:38.701000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [2/0] TRACE LOAD_FAST self [NullVariable(), GetAttrVariable()]
V0416 15:25:38.701000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [2/0] TRACE LOAD_CONST x [NullVariable(), GetAttrVariable(), LazyVariableTracker()]
V0416 15:25:38.701000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [2/0] TRACE LOAD_FAST x [NullVariable(), GetAttrVariable(), LazyVariableTracker(), ConstantVariable()]
V0416 15:25:38.701000 139970242536640 torch/_dynamo/symbolic_convert.py:784] [2/0] TRACE CALL 3 [NullVariable(), GetAttrVariable(), LazyVariableTracker(), ConstantVariable(), LazyVariableTracker()]
Eframes [('total', 3), ('ok', 2)]
inline_call []
stats [('calls_captured', 1), ('unique_graphs', 1)]

======================================================================
ERROR: test_object_setattr (__main__.MiscTests.test_object_setattr)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/PanXuehai/Projects/pytorch/torch/testing/_internal/common_utils.py", line 2741, in wrapper
    method(*args, **kwargs)
  File "/home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py", line 3095, in test_object_setattr
    obj22 = opt_fn2(x2.clone())
            ^^^^^^^^^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/eval_frame.py", line 410, in _fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py", line 3087, in fn2
    b = B(x)
        ^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 978, in catch_errors
    return callback(frame, cache_entry, hooks, frame_state, skip=1)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 818, in _convert_frame
    result = inner_convert(
             ^^^^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 411, in _convert_frame_assert
    return _compile(
           ^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_utils_internal.py", line 70, in wrapper_function
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/PanXuehai/Miniconda3/envs/pytorch-deps/lib/python3.12/contextlib.py", line 81, in inner
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 729, in _compile
    raise InternalTorchDynamoError(str(e)).with_traceback(
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 700, in _compile
    guarded_code = compile_inner(code, one_graph, hooks, transform)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/utils.py", line 266, in time_wrapper
    r = func(*args, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 568, in compile_inner
    out_code = transform_code_object(code, transform)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/bytecode_transformation.py", line 1113, in transform_code_object
    transformations(instructions, code_options)
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 173, in _fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 515, in transform
    tracer.run()
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2223, in run
    super().run()
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 879, in run
    while self.step():
          ^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 789, in step
    self.dispatch_table[inst.opcode](self, inst)
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 492, in wrapper
    return inner_fn(self, inst)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 1850, in CALL
    self.call_function(fn, args, kwargs)
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 729, in call_function
    self.push(fn.call_function(self, args, kwargs))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/variables/misc.py", line 611, in call_function
    return self.obj.call_method(tx, self.name, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/variables/builtin.py", line 959, in call_method
    obj = obj.realize()
          ^^^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/variables/lazy.py", line 57, in realize
    self._cache.realize()
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/variables/lazy.py", line 24, in realize
    self.vt = VariableBuilder(tx, self.source)(self.value)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/variables/builder.py", line 281, in __call__
    vt = self._wrap(value)
         ^^^^^^^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/variables/builder.py", line 780, in _wrap
    elif TorchCtxManagerClassVariable.is_matching_cls(value):
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/variables/torch.py", line 184, in is_matching_cls
    return hashable(value) and value in supported_ctx_manager_classes
           ^^^^^^^^^^^^^^^
  File "/home/PanXuehai/Projects/pytorch/torch/_dynamo/utils.py", line 435, in hashable
    hash(x)
  File "<string>", line 3, in __hash__
torch._dynamo.exc.InternalTorchDynamoError: 'B' object has no attribute 'x'

from user code:
   File "<string>", line 3, in __init__


You can suppress this exception and fall back to eager by setting:
    import torch._dynamo
    torch._dynamo.config.suppress_errors = True


To execute this test, run the following from the base repo dir:
     python test/dynamo/test_misc.py -k test_object_setattr

This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0

----------------------------------------------------------------------
Ran 1 test in 0.200s

FAILED (errors=1)
I0416 15:25:38.705000 139970242536640 torch/_dynamo/utils.py:324] TorchDynamo compilation metrics:
I0416 15:25:38.705000 139970242536640 torch/_dynamo/utils.py:324] Function, Runtimes (s)
V0416 15:25:38.705000 139970242536640 torch/fx/experimental/symbolic_shapes.py:106] lru_cache_stats evaluate_expr: CacheInfo(hits=0, misses=0, maxsize=256, currsize=0)
V0416 15:25:38.705000 139970242536640 torch/fx/experimental/symbolic_shapes.py:106] lru_cache_stats _simplify_floor_div: CacheInfo(hits=0, misses=0, maxsize=None, currsize=0)
V0416 15:25:38.705000 139970242536640 torch/fx/experimental/symbolic_shapes.py:106] lru_cache_stats _maybe_guard_rel: CacheInfo(hits=0, misses=0, maxsize=256, currsize=0)
V0416 15:25:38.705000 139970242536640 torch/fx/experimental/symbolic_shapes.py:106] lru_cache_stats _find: CacheInfo(hits=0, misses=0, maxsize=None, currsize=0)
V0416 15:25:38.705000 139970242536640 torch/fx/experimental/symbolic_shapes.py:106] lru_cache_stats has_hint: CacheInfo(hits=0, misses=0, maxsize=256, currsize=0)
V0416 15:25:38.705000 139970242536640 torch/fx/experimental/symbolic_shapes.py:106] lru_cache_stats size_hint: CacheInfo(hits=0, misses=0, maxsize=256, currsize=0)
V0416 15:25:38.705000 139970242536640 torch/fx/experimental/symbolic_shapes.py:106] lru_cache_stats simplify: CacheInfo(hits=0, misses=0, maxsize=None, currsize=0)
V0416 15:25:38.705000 139970242536640 torch/fx/experimental/symbolic_shapes.py:106] lru_cache_stats _update_divisible: CacheInfo(hits=0, misses=0, maxsize=None, currsize=0)
V0416 15:25:38.705000 139970242536640 torch/fx/experimental/symbolic_shapes.py:106] lru_cache_stats replace: CacheInfo(hits=0, misses=0, maxsize=None, currsize=0)
V0416 15:25:38.705000 139970242536640 torch/fx/experimental/symbolic_shapes.py:106] lru_cache_stats _maybe_evaluate_static: CacheInfo(hits=0, misses=0, maxsize=None, currsize=0)
V0416 15:25:38.705000 139970242536640 torch/fx/experimental/symbolic_shapes.py:106] lru_cache_stats safe_expand: CacheInfo(hits=0, misses=0, maxsize=256, currsize=0)
V0416 15:25:38.705000 139970242536640 torch/fx/experimental/symbolic_shapes.py:106] lru_cache_stats uninteresting_files: CacheInfo(hits=0, misses=0, maxsize=None, currsize=0)
```

</details>

Reply:
The class B definition:

```python
@dataclass(frozen=True)
class B:
    x: torch.Tensor
```

is equivalent to:

```python
__dataclass_builtins_object__ = object

class B:
    def __init__(self, x: torch.Tensor) -> None:
        __dataclass_builtins_object__.__setattr__(self, 'x', x)

    def __hash__(self):
        return hash((self.x,))

    ...
```